internal/storage/storageutil/client_helper_test.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storageutil
+
+import (
+	"net/url"
+	"testing"
+
+	. "github.com/jacobsa/ogletest"
+)
+
+func TestClientHelper(t *testing.T) { RunTests(t) }
+
+type clientHelperTest struct {
+}
+
+func init() { RegisterTestSuite(&clientHelperTest{}) }
+
+func (t *clientHelperTest) TestIsGCSProdHostnameWithCustomName() {
+	url, err := url.Parse(CustomEndpoint)
+	AssertEq(nil, err)
+
+	res := IsProdEndpoint(url)
+
+	ExpectFalse(res)
+}
+
+func (t *clientHelperTest) TestIsGCSProdHostnameEndpoint() {
+	// GCSFuse assumes prod, if we specify endpoint as nil.
+	var url *url.URL
+
+	res := IsProdEndpoint(url)
+
+	ExpectTrue(res)
+}
+
+func (t *clientHelperTest) TestCreateTokenSrcWithCustomEndpoint() {
+	url, err := url.Parse(CustomEndpoint)
+	AssertEq(nil, err)
+
```

Comment:
[go/unit-testing-practices#structure](http://goto.google.com/unit-testing-practices#structure)

No new lines in arrange block